spring-kafka/src/main/java/org/springframework/kafka/listener/ConsumerAwareRecordInterceptor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import org.springframework.lang.Nullable;
+
+/**
+ * A {@link RecordInterceptor} that has access to the {@link Consumer}.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Gary Russell
+ * @since 2.7
+ *
+ */
+public interface ConsumerAwareRecordInterceptor<K, V> extends RecordInterceptor<K, V> {
```

Comment:
`@FunctionalInterface` ?

I guess we can live without this intermediary since all the methods are present in the `RecordInterceptor`, but I believe you have introduced it for end-user convenience.